database/src/main/scala/cromwell/database/SqlDatabase.scala

```diff
@@ -213,4 +216,17 @@ trait SqlDatabase extends AutoCloseable {
                                     index: Option[Int],
                                     attempt: Int)
                                    (implicit ec: ExecutionContext): Future[Seq[Metadatum]]
+
+  /**
+    * Retrieves all summarizable metadata satisfying the specified criteria.
+    *
+    * @param startMetadataId        The minimum ID an entry in `METADATA_JOURNAL` must have to be examined for summary.
+    * @param startMetadataTimestamp An optional timestamp.  If specified, a metadatum must have a timestamp greater than or equal to this value.
+    * @return A `Future` with the maximum ID value of the metadata summarized by the invocation of this method.
+    */
+  protected def refreshMetadataSummaries(startMetadataId: Long, startMetadataTimestamp: Option[OffsetDateTime])
+                                        (implicit ec: ExecutionContext): Future[Long]
+
+  protected def getStatus(id: WorkflowId)
```

Comment:
Remove the `core` dependency, and pass in the `workflowUuid: String`. IMHO there's too much other stuff in `core` that will leak deep into the `database`, again.